src/Runner.Worker/JobHookProvider.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System.Linq;
+using GitHub.Runner.Common;
+using GitHub.Runner.Common.Util;
+using GitHub.Runner.Sdk;
+using GitHub.Runner.Worker.Handlers;
+
+namespace GitHub.Runner.Worker
+{
+    [ServiceLocator(Default = typeof(JobHookProvider))]
+    public interface IJobHookProvider : IRunnerService
+    {
+        Task RunHook(IExecutionContext executionContext, object data);
+    }
+
+    public class JobHookData
+    {
+        public string Path {get; private set;}
+        public ActionRunStage Stage {get; private set;}
+        public string DisplayName {get; private set;}
+
+        public JobHookData(ActionRunStage stage, string path, string displayName)
+        {
+            Path = path;
+            Stage = stage;
+            DisplayName = displayName;
+        }
+    }
+
+    public class JobHookProvider : RunnerService, IJobHookProvider
+    {
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+        }
+
+        public async Task RunHook(IExecutionContext executionContext, object data)
+        {
+            // Get Inputs
+            var hookData = data as JobHookData;
+            ArgUtil.NotNull(hookData, nameof(JobHookData));
+
+            // Log to users so that they know how this step was injected
+            executionContext.Output($"A '{hookData.DisplayName}' has been configured by the Self Hosted Runner Administrator");
```

Comment:
```suggestion
            executionContext.Output($"A '{hookData.DisplayName}' has been configured by the self-hosted runner administrator");
```

We don't generally use Title Case, and we prefer "self-hosted" in comms.  But I also think that hooks should have a name and a title (or displayname, don't care about what it's called in the code).

<img width="679" alt="Screen Shot 2022-03-16 at 15 30 19" src="https://user-images.githubusercontent.com/1130014/158674842-fe46ead4-271c-4ee3-93f6-30200fbdf15b.png">

I'd like this to read something more like: "A job started hook has been configured by the self-hosted runner administrator".